turbopack/crates/turbo-tasks-backend/src/backing_storage.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use turbo_tasks::{backend::CachedTaskType, TaskId};
+
+use crate::{
+    backend::{AnyOperation, TaskDataCategory},
+    data::{CachedDataItem, CachedDataUpdate},
+    utils::chunked_vec::ChunkedVec,
+};
+
+#[derive(Clone, Copy)]
+pub struct ReadTransaction(pub *const ());
```

Comment:
As it's currently using `Arc<dyn BackingStorage + Sync + Send>` we can't have associated types on that, but I think it might be possible when `TurboTasksBackendInner` is generic `TurboTasksBackendInner<B: BackingStorage>`. That's requires making a ton of code generic. I'll try to do that, but as follow-up PR